test/proxy/Clones.t.sol

```diff
@@ -15,4 +19,42 @@ contract ClonesTest is Test {
         }
         assertEq(spillage, bytes32(0));
     }
+
+    function testCloneDirty() external {
+        address cloneClean = Clones.clone(address(this));
+        address cloneDirty = Clones.clone(_dirty(address(this)));
+
+        // both clones have the same code
+        assertEq(keccak256(cloneClean.code), keccak256(cloneDirty.code));
+
+        // both clones behave as expected
+        assertEq(ClonesTest(cloneClean).getNumber(), this.getNumber());
+        assertEq(ClonesTest(cloneDirty).getNumber(), this.getNumber());
+    }
+
+    function testCloneDeterministicDirty() external {
+        address cloneClean = Clones.cloneDeterministic(address(this), bytes32(block.prevrandao));
+        address cloneDirty = Clones.cloneDeterministic(_dirty(address(this)), bytes32(~block.prevrandao));
+
+        // both clones have the same code
+        assertEq(keccak256(cloneClean.code), keccak256(cloneDirty.code));
+
+        // both clones behave as expected
+        assertEq(ClonesTest(cloneClean).getNumber(), this.getNumber());
+        assertEq(ClonesTest(cloneDirty).getNumber(), this.getNumber());
+    }
+
+    function testPredictDeterministicAddressDirty(bytes32 salt) external {
+        address predictClean = Clones.predictDeterministicAddress(address(this), salt);
+        address predictDirty = Clones.predictDeterministicAddress(_dirty(address(this)), salt);
+
+        //prediction should be similar
+        assertEq(predictClean, predictDirty);
+    }
```

Comment:
What is this testing? Since the address is still of `address` type it'll pass the test even if dirty.
I have the sense this is testing the same as `testSymbolicPredictDeterministicAddressSpillage`.

Reply:
This tests that `predictDeterministicAddress` returns the same value if the input address (implementation) is dirty.

Reply:
Right, the other one is just testing the spillage. Clearer now, solving